Marlin/src/module/probe.h

```diff
@@ -199,7 +199,7 @@ class Probe {
   static void move_z_after_probing() {
     DEBUG_SECTION(mzah, "move_z_after_probing", DEBUGGING(LEVELING));
     #ifdef Z_AFTER_PROBING
-      do_z_clearance(Z_AFTER_PROBING, true, true); // Move down still permitted
+      do_z_clearance(Z_AFTER_PROBING, true); // Move down still permitted
```

Comment:
With this change, Z_AFTER_PROBING is a fixed, absolute Z hight of the nozzle after probing. I like to have these fixed endpoints for commands. But we have to keep in mind that this is a change that breaks backwards-compatibillity. Currently, users can freely adjust nozzle-to-probe offset becausse the clearance after probing will be adjusted. With this change, they must keep probe.offset.z > (- Z_AFTER_PROBING).

Reply:
> With this change, Z_AFTER_PROBING is a fixed, absolute Z hight of the nozzle after probing. I like to have these fixed endpoints for G-code commands. But we have to keep in mind that this is a change that breaks backwards-compatibillity. Currently, users can freely adjust nozzle-to-probe offset becausse the clearance after probing will be adjusted. With this change, they must keep probe.offset.z > (- Z_AFTER_PROBING).

This has nothing really to do with "backwards-compatibility", especially reverting back to older code. All this does is stop adding/subtracting the probe's Z offset from Homing / Z clearance - which in turn gives the *intended* **true** height.  
Clearance before and after changes is not greatly affected. Z clearance height is true after changes, and before changes it is +/- probe Z offset.

I'm trying to see what you mean either way...:
>Currently, users can freely adjust nozzle-to-probe offset becausse the clearance after probing will be adjusted. 

Clearance after probing is not affected negatively, but it is beneficial.

- If  <b>`+`</b>`probe.offset.z` ( > 0 ):
- - This brings the nozzle further **UP**

- If  <b>`-`</b>`probe.offset.z` ( < 0 ):
- - This brings the nozzle further **DOWN**

---

### See the diagram in my related issue https://github.com/MarlinFirmware/Marlin/issues/27294 :
![zoffsetHoming](https://github.com/user-attachments/assets/05d353cd-3a9d-46a3-b1b2-31ae6b01f6e0)


Let's take what you say if `probe.offset.z > (- Z_AFTER_PROBING)`, according to the diagram above:

#### Homing will *not* bring the Z_AFTER_PROBING to the desired +10.00 above the bed. With the current logic,  Z Offset +/positive gets subtracted, -/negative gets added (`zdest -= probe.offset.z`).

- *This* is represented in the **Middle**.  Let's say you need a positive Z Offset of +10.00 in order for the Nozzle to meet the bed's surface, +10.00 gets subtracted from +10.00 of `Z_AFTER_PROBING` (assuming that is 10), so you're left 0.00 height at Home which is AT the bed surface, not above.

- The **Left** side represents a Z Offset of 0.00 (None/default). So homing may leave it +10 above the bed, but when Z=0 this will cause the Nozzle to go below the bed surface. That is why a positive Z Offset is needed, since it raises the Nozzle to the correct bed height.

- The **Right** side represents the opposite, a negative Z Offset will bring the Home position +20 above the bed, since `probe.offset.z = -10.00; zdest = zdest - (-10.00);`.  
And since a negative Z Offset *lowers* the nozzle when Z=0 it goes an additional 10 below the already 10 below, so 20 below the bed (or -20).